tests/sdp-negotiation.rs

```diff
@@ -0,0 +1,187 @@
+mod common;
+use common::init_log;
+use str0m::format::Codec;
+use str0m::format::CodecSpec;
+use str0m::format::FormatParams;
+use str0m::format::PayloadParams;
+use str0m::media::Direction;
+use str0m::media::MediaKind;
+use str0m::Rtc;
+use tracing::info_span;
+
+fn with_config(params: &[PayloadParams]) -> Rtc {
+    let mut b = Rtc::builder().clear_codecs();
+    let config = b.codec_config();
+    for p in params {
+        config.add_config(
+            p.pt(),
+            p.resend(),
+            p.spec().codec,
+            p.spec().clock_rate,
+            p.spec().channels,
+            p.spec().format,
+        );
+    }
+    b.build()
+}
+
+#[test]
+pub fn change_pt_sendrecv() {
+    init_log();
+
+    let (l, r) = negotiate(
+        //
+        &[opus(100, 2)],
+        &[opus(102, 2)],
+        MediaKind::Audio,
+        Direction::SendRecv,
+    );
+
+    // For sendrecv, both sides state their mandatory receive parameters. I.e.
+    // L -> R must be 102
+    // R -> L must be 100
+
+    assert_eq!(
+        &[opus(102, 2)],
+        &**l.codec_config_tx(),
+        "L sends to R on PT 102"
+    );
+    assert_eq!(
+        &[opus(100, 2)],
+        &**l.codec_config_rx(),
+        "L receives from R on PT 100"
+    );
+    assert_eq!(
+        &[opus(102, 2)],
+        &**r.codec_config_rx(),
+        "R send to L on PT 100"
+    );
+    assert_eq!(
+        &[opus(102, 2)],
+        &**r.codec_config_rx(),
+        "R receives from L on PT 102"
+    );
+}
+
+#[test]
+pub fn change_pt_sendonly() {
+    init_log();
+
+    let (l, r) = negotiate(
+        //
+        &[opus(100, 2)],
+        &[opus(102, 2)],
+        MediaKind::Audio,
+        Direction::SendOnly,
+    );
+
+    // For sendonly only the ANSWER from R is mandatory. The OFFER contains suggestions.
+    // L -> R must be 102
+    // R -> L can be anything, but str0m tries to match the remote, so also 102
```

Comment:
Is this true? Doesn't the pt for L remain unbound until an offer is made with a `recvonly` or `sendrecv` direction? The value 100 here is just a suggestion for what it wants to send i.e. `l.codec_config_tx` should be 102 and `r.codec_config_rx` should be `102`.

Reply:
I guess 🤔